src/sleeper.rs

```diff
@@ -0,0 +1,14 @@
+use std::time::Duration;
+
+pub trait Sleeper {
+    fn sleep(duration: Duration) -> impl std::future::Future<Output = ()> + Send;
+}
```

Comment:
I think we'd need to add `#[async_trait]` for this. As far as I know "return position `impl Trait` in traits" didn't stabilize until rust 1.75

Reply:
@ValuedMammal yes you're right, should I just use `async_trait` or use [this build dep](https://docs.rs/version_check/latest/version_check/) to conditionally use `async_trait` on rust versions lower than 1.75 only?

Reply:
I guess I vote for `async_trait`

Reply:
sure i'll add it, and we can remove when MSRV gets bumped, when/how does that happen?

Reply:
@ValuedMammal  done